archive-cleaner

```diff
@@ -120,8 +136,8 @@ def main(repo=DEFAULT_SOURCE_DIR, archive=DEFAULT_ARCHIVE_DIR, keep_years=DEFAUL
     years = list(filter(lambda year: year <= keep_until,
                         map(lambda path: int(basename(path)), glob(join(archive, 'repos', '2*')))))
     purge_paths = [path for paths in [glob(join(archive, 'repos', f'{year}', '*', '*')) for year in years] for path in paths]
-    purge_paths = [join(path, '*', 'os', '*', f'*{PKG_EXT}') for path in purge_paths] + \
-                  [join(path, 'pool', '*', f'*{PKG_EXT}') for path in purge_paths]
+    purge_paths = [join(path, '*', 'os', '*', f'*{ext}') for ext in PKG_EXTENSIONS for path in purge_paths] + \
```

Comment:
I think a similar approach as described above works better than submitting tons of glob tasks to the pool. Just reduce the PKG_EXT park to take any kind of .pkg. files and filter out signatures.

Reply:
Would *.pkg.* work as a glob here then?

Reply:
```
    purge_paths = [join(path, '*', 'os', '*', '*.pkg.*') for path in purge_paths] + \
                  [join(path, 'pool', '*', '*.pkg.*') for path in purge_paths]

    with Pool(processes=processes) as pool:
        results = pool.map(glob, purge_paths)

    all_files = [path for paths in results for path in paths if SIG_EXT not in path]
```
something like this I mean

Reply:
yeah, that should do it. however, you will need to filter out the results of pool.map to not contain .sig ending files

Reply:
yeah it should be doing it in the last line when i assign all_files